### PR TITLE
Meta+Tests: Reuse existing test sources lists instead of globbing

### PR DESCRIPTION
**Pull request by AtkinsSJ**
image

As noted [here](https://github.com/SerenityOS/serenity/pull/16934#pullrequestreview-1249036537), globbing files causes issues. So, let's do less of it!

This also includes some minor clean-up of the tests to make things work.

cc: @ADKaster

### CODE REVIEW

File: Tests/LibMarkdown/CMakeLists.txt [hunk 1]
```diff
@@ -8,3 +8,7 @@ set(TEST_SOURCES
 foreach(source IN LISTS TEST_SOURCES)
     serenity_test("${source}" LibMarkdown LIBS LibMarkdown)
 endforeach()
+
+if (BUILD_LAGOM)
+set_tests_properties(TestCommonmark PROPERTIES DISABLED YES)
+endif()
```
Comment (AtkinsSJ) [inline Tests/LibMarkdown/CMakeLists.txt:14]: This is a bit funky, but as far as I can tell is the only way to do this. You can't call set_tests_properties() from another file. https://gitlab.kitware.com/cmake/cmake/-/issues/22813

`BUILD_LAGOM` is maybe not the best way of detecting that we should disable the tests, but it works. :man_shrugging:

**Comment by AtkinsSJ** [comment]
Changes: Make sure to copy LibPDF's test files to `/usr/Tests/LibPDF` so that the CI tests work.

File: Tests/LibMarkdown/CMakeLists.txt (hunk 1 above)
Comment (ADKaster) [reply Tests/LibMarkdown/CMakeLists.txt:14]: if(NOT SERENITYOS) would be better, but not sure what that does for building serenity on serenity

File: Meta/Lagom/CMakeLists.txt [hunk 2]
```diff
@@ -259,6 +259,11 @@ function(lagom_test source)
     )
 endfunction()
 
+function(serenity_test test_src sub_dir)
+    cmake_parse_arguments(PARSE_ARGV 2 SERENITY_TEST "MAIN_ALREADY_DEFINED" "CUSTOM_MAIN" "LIBS")
```
Comment (ADKaster) [inline Meta/Lagom/CMakeLists.txt:263]: Could use a FIXME here to pass along main already defined and custom main when we get to porting tests that use it.

File: Meta/Lagom/CMakeLists.txt [hunk 3]
```diff
@@ -558,118 +563,61 @@ if (BUILD_LAGOM)
 
         # LibTest tests from Tests/
         # AK
-        file(GLOB AK_TEST_SOURCES CONFIGURE_DEPENDS "../../Tests/AK/*.cpp")
-        foreach(source ${AK_TEST_SOURCES})
-            lagom_test(${source} WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../../Tests/AK)
-        endforeach()
-        target_link_libraries(TestString PRIVATE LibUnicode)
+        add_serenity_subdirectory(Tests/AK)
```
Comment (ADKaster) [inline Meta/Lagom/CMakeLists.txt:566]: 🤔 
```
set(TEST_DIRECTORIES
    AK
    LibAudio
   LibFoo
   LibBar
  ...
)
if (ENABLE_LAGOM_LIBWEB)
   list(APPEND TEST_DIRECTORIES LibWeb)
endif()

foreach ( dir IN LISTS TEST_DIRECTORIES)
   add_serenity_subdirectory("Tests/${dir}")
endforeach()
```

**Review by ADKaster — COMMENTED** [review]
Nice 🙂👍. Two suggestions.

I'm not a fan of leaking BUILD_LAGOM outside of Lagom/CMakeLists.txt

Perhaps we could get fancy by checking if testing is enabled later.

File: Meta/Lagom/CMakeLists.txt (hunk 3 above)
Comment (AtkinsSJ) [reply Meta/Lagom/CMakeLists.txt]: Oooooh. Fancy!

**Comment by AtkinsSJ** [comment]
Push 1: Addressing comments.
Push 2: Rebase to fix a small conflict.

File: Tests/LibTTF/CMakeLists.txt [hunk 4]
```diff
@@ -3,5 +3,5 @@ set(TEST_SOURCES
 )
 
 foreach(source IN LISTS TEST_SOURCES)
-    serenity_test("${source}" LibGfx)
+    serenity_test("${source}" LibTTF LIBS LibGfx)
```
Comment (linusg) [inline Tests/LibTTF/CMakeLists.txt:6]: (first commit) did you mean to say "it's tidier" in the commit message?

File: Tests/LibMarkdown/CMakeLists.txt [hunk 5]
```diff
@@ -8,3 +8,7 @@ set(TEST_SOURCES
 foreach(source IN LISTS TEST_SOURCES)
     serenity_test("${source}" LibMarkdown LIBS LibMarkdown)
 endforeach()
+
+if (BUILD_LAGOM)
+set_tests_properties(TestCommonmark PROPERTIES DISABLED YES)
```
Comment (linusg) [inline Tests/LibMarkdown/CMakeLists.txt:13]: Missing indentation

File: Tests/LibTTF/CMakeLists.txt (hunk 4 above)
Comment (AtkinsSJ) [reply Tests/LibTTF/CMakeLists.txt:6]: Yes I did!

**Comment by AtkinsSJ** [comment]
Thanks Linus! Rebased, and then addressed your comments.

File: Meta/Lagom/CMakeLists.txt [hunk 6]
```diff
@@ -577,104 +595,9 @@ if (BUILD_LAGOM)
...
         endif()
 
-        # Crypto
-        file(GLOB LIBCRYPTO_TESTS CONFIGURE_DEPENDS "../../Tests/LibCrypto/*.cpp")
-        foreach(source ${LIBCRYPTO_TESTS})
-            lagom_test(${source} LIBS LibCrypto)
-        endforeach()
-
-        # Compress
-        file(COPY "${SERENITY_PROJECT_ROOT}/Tests/LibCompress/brotli-test-files" DESTINATION "./")
```
[the remark this answers is not in the document]
Comment (nico) [reply Meta/Lagom/CMakeLists.txt:587]: (Follow-up from discord: The CWD when running a lagom test is now supposed to be the directory containing these files. See also #17261 and #17263.)

**Pull request merged**